earth_enterprise/src/fusion/autoingest/AssetFactory.h

```diff
@@ -261,274 +238,189 @@ namespace AssetFactory
                cachedinputs_);
   }
 
-  template <class MutableDerivedVersionHandleType, class Version, class ConfigType>
-  MutableDerivedVersionHandleType FindMakeAndUpdateSubAsset(const std::string& parentAssetRef,
+  template <class AssetType>
+  typename AssetType::MutableVersionD FindMakeAndUpdateSubAsset(const std::string& parentAssetRef,
                                                             const std::string& basename,
                                                             const std::vector<SharedString>& inputs_,
                                                             const khMetaData& meta_,
-                                                            const ConfigType& config,
-                                                            const std::vector<Version>& cachedinputs_)
+                                                            const typename AssetType::Config& config,
+                                                            const std::vector<typename AssetType::Version::Base>& cachedinputs_)
   {
-      using Impl = typename MutableDerivedVersionHandleType::Impl;
       auto ref_ = AssetDefs::SubAssetName(parentAssetRef, basename,
-                                          Impl::EXPECTED_TYPE, Impl::EXPECTED_SUBTYPE);
-      return FindMakeAndUpdate<MutableDerivedVersionHandleType, Version, ConfigType>
+                                          AssetType::TYPE, AssetType::SUBTYPE);
+      return FindMakeAndUpdate<AssetType>
              (ref_,
               inputs_,
               meta_,
               config,
               cachedinputs_);
   }
 
-
-  template <class MutableDerivedVersionHandleType,
-            class ConfigType, class VersionDType>
-  MutableDerivedVersionHandleType ReuseOrMakeAndUpdate(const std::string& ref_,
```

Comment:
I copied over all the ReuseOrMakeAndUpdate* functions from earlier versions, which was the fix to one of the issues. Unfortunately it makes the diff hard to read.